src/components/pages/Sources/SourceDetails/SourceSettings/General/Authentication.tsx

```diff
@@ -31,31 +36,34 @@ const Authentication: React.FC = () => {
   const token = repository?.tokenSecret?.name || '';
   const username = repository?.usernameSecret?.name || '';
 
-  const [form] = Form.useForm();
+  const [form] = Form.useForm<AuthenticationFormValues>();
 
-  const onFinish = (values: {token: string; username: string}) => {
+  const onFinish = (values: AuthenticationFormValues) => {
     if (!source) {
       notificationCall('failed', 'Something went wrong.');
-      return;
+    } else {
```

Comment:
Why not just `return` as it was before, to keep it as a simple control check?

Reply:
Just don't like early return pattern, explained it a bit upper. Do you feel we should use it as a simple control check? If `if` works, `else` does not 😅